src/components/inputs/select/dropdown.component.tsx

```diff
@@ -11,16 +11,36 @@ import { FormContext } from '../../../form-context';
 import { type FormFieldProps } from '../../../types';
 import RequiredFieldLabel from '../../required-field-label/required-field-label.component';
 import styles from './dropdown.scss';
+import { getControlTemplate } from '../../../registry/inbuilt-components/control-templates';
+import { getRegisteredDataSource } from '../../../registry/registry';
 
 const Dropdown: React.FC<FormFieldProps> = ({ question, onChange, handler, previousValue }) => {
   const { t } = useTranslation();
   const [field, meta] = useField(question.id);
   const { setFieldValue, encounterContext, layoutType, workspaceLayout, fields } = React.useContext(FormContext);
+  const [datasourceItems, setDatasourceItems] = React.useState([]);
   const [items, setItems] = React.useState([]);
   const [errors, setErrors] = useState([]);
   const isFieldRequiredError = useMemo(() => errors[0]?.errCode == fieldRequiredErrCode, [errors]);
   const [warnings, setWarnings] = useState([]);
 
+  useEffect(() => {
+    const template = getControlTemplate(question.questionOptions.rendering);
+    // check if the question has a control template as has a datasource name
+    if (template?.datasource?.name) {
+      getRegisteredDataSource(template.datasource.name).then((ds) => {
+        ds.fetchData(question.questionOptions.concept).then((dataItems) => {
+          setItems(dataItems.map((item) => item.uuid));
+          setDatasourceItems(dataItems.map((item) => ({ value: item.uuid, label: item.display })));
+        });
+      });
+    } else {
+      setItems(
+        question.questionOptions.answers.filter((answer) => !answer.isHidden).map((item) => item.value || item.concept),
+      );
+    }
+  }, [question.questionOptions]);
```

Comment:
@usamaidrsk The ui-select-extended already was designed to work with datasources. We use the same approach for encounter providers, drugs etc. You simply want to add an entry [here](https://github.com/openmrs/openmrs-form-engine-lib/blob/main/src/registry/inbuilt-components/control-templates.ts) associating the datasource to the derived rendering type. That being said, you don't need to touch this file.